src/main/java/Aly.java

```diff
@@ -0,0 +1,242 @@
+                printLine();
+                echo();
+                break;
+            case "2":
+                System.out.println("Enter 0 to return to main menu anytime!");
+                printLine();
+                list(tasks);
+                break;
+            case "3":
+                System.out.println("Enter 0 to return to main menu anytime!");
+                printLine();
+                markTask(tasks);
+                break;
+            default:
+                System.out.println("Can read instructions anot? Try again!");
+                break;
+            }
+        }
+        exit();
+    }
+
+    private static void exit() {
+        System.out.println("Bye! I'm going to eat MacDonalds now!");
+        printLine();
+        System.exit(0);
+    }
+
+    private static void echo() {
+        boolean isExit = false;
+        while (!isExit) {
+            System.out.println("Waiting for Input...");
+            String line = in.nextLine().trim();
+            if (line.equals("0")) {
+                isExit = true;
+                System.out.println("Returning to main menu!");
+            } else {
+                printLine();
+                System.out.println(line);
+                printLine();
+            }
+        }
+    }
+
+    private static void list(Task[] listItems) {
+        boolean isExit = false;
+        int index = 0;
+        index = getIndex(listItems, index);
+        while (!isExit) {
+            System.out.println("Enter 'list' to see your list of tasks, 'todo'/'deadline'/'event' to add that respective type of task, '0' to exit");
+            String input = in.nextLine().trim();
+            String firstWord = input.split(" ")[0];
+            String task = input.replace(firstWord, "");
+
+            switch (firstWord) {
+            case "list":
+                listTasks(listItems);
+                break;
+            case "todo":
+                index = addTodo(listItems, task, index);
+                break;
+            case "deadline":
+                index = addDeadline(listItems, task, index);
+                break;
+            case "event":
+                index = addEvent(listItems, task, index);
+                break;
+            case "0":
+                isExit = true;
+                System.out.println("Returning to main menu!");
+                break;
+            default:
+                printLine();
+                System.out.println("Can read instructions anot? Try again!");
+                printLine();
+                break;
+            }
+        }
+    }
+
+    private static void listTasks(Task[] listItems) {
+        printLine();
+        int count = 1;
+        System.out.println("Your task list:");
+        for (Task listItem : listItems) {
+            if (listItem == null) {
+                break;
+            }
+            System.out.println(count + "." + listItem);
+            count += 1;
```

Comment:
Perhaps "count++" could be used in place of "count += 1", and same for "index" below?